Autocoders/MagicDrawCompPlugin/src/gov/nasa/jpl/componentaction/LoadIDConfig.java

```diff
@@ -167,7 +167,7 @@ public static void process(Project proj, File pluginDir) throws IOException {
 	 * @throws LoadIDException
 	 */
 	public static void changeModel(Element root , Project proj ,  Map<String , configItem> configMap) throws LoadIDException{
-		Collection<Element> coll = (Collection<Element>)ModelHelper.getElementsOfType(root, null, true, true);
+		Collection<Element> collection = (Collection<Element>)ModelHelper.getElementsOfType(root, null, true, true);
```

Comment:
`col` is often an abbreviation for `column`. There's nothing technically wrong w/ `coll`, but it risks confusion, and Java encourages long variable names, such as `componentList` and `componentInstanceStereotype` below:

Reply:
Did you try building this change?

Reply:
Yes

```java
nasa/fprime/Autocoders/MagicDrawCompPlugin % make -f classes.mk
mkdir -p classes
javac -classpath lib/commons-cli-1.3.1.jar:lib/velocity-1.6.2-dep.jar:lib/md.jar:lib/bundles/com.nomagic.magicdraw.foundation_2.0.0.201803050637.jar:lib/bundles/org.eclipse.emf.common_2.10.1.v20140901-1043.jar:lib/bundles/javax.jmi_1.0.0.201803050637/javax_jmi-1_0-fr.jar:lib/bundles/org.junit_4.11.0.v201303080030/junit.jar:lib/bundles/com.nomagic.magicdraw.uml2_2.5.0.201803050637.jar:lib/bundles/org.eclipse.emf.ecore_2.10.1.v20140901-1043.jar:lib/md_api.jar: -d classes -encoding UTF8 src/gov/nasa/jpl/componentaction/ISFComponent.java src/gov/nasa/jpl/componentaction/IsfTopAction.java src/gov/nasa/jpl/componentaction/ProcessISFTopology.java src/gov/nasa/jpl/componentaction/MpmcsErrorStream.java src/gov/nasa/jpl/componentaction/ISFPort.java src/gov/nasa/jpl/componentaction/isfxmlwriter/IsfPortXmlWriter.java src/gov/nasa/jpl/componentaction/isfxmlwriter/IsfSubXmlWriter.java src/gov/nasa/jpl/componentaction/isfxmlwriter/IsfCompXmlWriter.java src/gov/nasa/jpl/componentaction/IsfXmlAutocoderAction.java src/gov/nasa/jpl/componentaction/LoadIDException.java src/gov/nasa/jpl/componentaction/Utils.java src/gov/nasa/jpl/componentaction/LoadIDConfigAction.java src/gov/nasa/jpl/componentaction/IsfXmlAction.java src/gov/nasa/jpl/componentaction/LoadIDConfig.java src/gov/nasa/jpl/componentaction/Partition.java src/gov/nasa/jpl/componentaction/IsfComponentAction.java src/gov/nasa/jpl/componentaction/ProcessISFProject.java src/gov/nasa/jpl/componentaction/ConnectorException.java src/gov/nasa/jpl/componentaction/ISFSubsystem.java src/gov/nasa/jpl/componentaction/IsfAbout.java src/gov/nasa/jpl/componentaction/PortException.java src/gov/nasa/jpl/componentaction/ComponentException.java src/gov/nasa/jpl/componentaction/MainMenuConfigurator.java src/gov/nasa/jpl/componentaction/TestISFExceptions.java
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: Some input files use unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
nasa/fprime/Autocoders/MagicDrawCompPlugin % javap classes/gov/nasa/jpl/componentaction/LoadIDConfig.class
Compiled from "LoadIDConfig.java"
public class gov.nasa.jpl.componentaction.LoadIDConfig {
  public static void process(com.nomagic.magicdraw.core.Project, java.io.File) throws java.io.IOException;
  public static void changeModel(com.nomagic.uml2.ext.magicdraw.classes.mdkernel.Element, com.nomagic.magicdraw.core.Project, java.util.Map<java.lang.String, gov.nasa.jpl.componentaction.LoadIDConfig$configItem>) throws gov.nasa.jpl.componentaction.LoadIDException;
  public static java.util.Map<java.lang.String, gov.nasa.jpl.componentaction.LoadIDConfig$configItem> validateAndProcessConfig(java.io.File) throws gov.nasa.jpl.componentaction.LoadIDException;
  public static java.io.File getConfigFile(java.io.File) throws gov.nasa.jpl.componentaction.LoadIDException;
}
nasa/fprime/Autocoders/MagicDrawCompPlugin % javap classes/gov/nasa/jpl/componentaction/ProcessISFTopology.class
Compiled from "ProcessISFTopology.java"
public class gov.nasa.jpl.componentaction.ProcessISFTopology {
  public static java.util.HashSet<gov.nasa.jpl.componentaction.ISFSubsystem$physicalConnectionType> physicalConnectionSet;
  public static java.util.HashMap<java.lang.String, gov.nasa.jpl.componentaction.ISFSubsystem$componentType> componentMap;
  public static java.util.ArrayList<gov.nasa.jpl.componentaction.ISFSubsystem$physicalConnectionType> physicalConnectionList;
  public static java.util.Map<java.lang.String, java.util.ArrayList<java.lang.String[]>> subsystemOwners;
  public static java.util.Map<java.lang.String, java.lang.Integer> subsystemInstances;
  public static java.util.Map<java.lang.String, java.util.ArrayList<java.lang.String>> uniqueSubsystemInstanceNames;
  public static void process(com.nomagic.magicdraw.core.Project, java.io.File) throws java.io.IOException;
  public static int generateNewMultiplicity(com.nomagic.uml2.ext.magicdraw.compositestructures.mdinternalstructures.Connector, int, int) throws gov.nasa.jpl.componentaction.ConnectorException;
}
```

n.b. the [README](https://github.com/nasa/fprime/blob/c30c61a6cb0f63d70d29c04ac31a60d53147947a/Autocoders/MagicDrawCompPlugin/README#L13) in the directory indicates that `make` should _just work_, it doesn't as there's no `Makefile`.

`classes.mk` works.

Also, `gen_classes_mk` generates a file that is conceptually equivalent to `classes.mk`, but the output's order isn't equivalent (the `.java` files are ordered differently, and the `.jar` files are ordered differently).

(I don't have MagicDraw, but I could get a demo if necessary.)